DCS_pcall.lua

```diff
@@ -17,127 +34,1227 @@ DCS = {
+                return _object:getVelocity();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        inAir = function(_object)
+            local _status, _response = pcall(function()
+                return _object:inAir();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end
+    },
+
+    ScenaryObject = {}, -- TODO ScenaryObject not started
+
+    Spot = {
+        createLaser = function(_source, _localRef, _point, _laseCode)
+            local _status, _response = pcall(function()
+                return DCS.proxy["Spot"].createLaser(_source, _localRef, _point, _laseCode);
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        createInfraRed = function(_source, _localRef, _point)
+            local _status, _response = pcall(function()
+                return DCS.proxy["Spot"].createInfraRed(_source, _localRef, _point);
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        destroy = function(_spot)
+            local _status, _ = pcall(function()
+                _spot:destroy();
+            end)
+            if not _status then
+                -- TODO Need debug logging added here
+            end
+            return nil;
+        end,
+
+        getCategory = function(_spot)
+            local _status, _response = pcall(function()
+                return _spot:getCategory();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        getPoint = function(_spot)
+            local _status, _response = pcall(function()
+                return _spot:getPoint();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        setPoint = function(_spot, _vec3)
+            local _status, _ = pcall(function()
+                _spot:setPoint(_vec3);
+            end)
+            if not _status then
+                -- TODO Need debug logging added here
+            end
+            return nil;
+        end,
+
+        getCode = function(_spot)
+            local _status, _response = pcall(function()
+                return DCS.proxy["Spot"].getCode(_spot);
```

Comment:
Should this one be `_spot:getCode()` instead of using the proxy? Or is it only available as a static in Hoggit documentation?